Example/Tests/GiniScreenAPICoordinatorTests.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  GiniScreenAPICoordinatorTests.swift
+//  GiniVision_Tests
+//
+//  Created by Enrique del Pozo Gómez on 3/8/18.
+//  Copyright © 2018 Gini GmbH. All rights reserved.
+//
+
+import XCTest
+@testable import GiniVision
+
+final class GiniScreenAPICoordinatorTests: XCTestCase {
+    
+    var coordinator: GiniScreenAPICoordinator!
+    let giniConfiguration = GiniConfiguration()
+    let delegate = GiniVisionDelegateMock()
+    
+    override func setUp() {
+        super.setUp()
+        giniConfiguration.openWithEnabled = true
+        coordinator = GiniScreenAPICoordinator(withDelegate: delegate, giniConfiguration: giniConfiguration)
+    }
+    
+    func testNavControllerCountAfterStartWithoutDocuments() {
+        let rootViewController = coordinator.start(withDocuments: nil)
+        _ = rootViewController.view
+        let screenNavigator = rootViewController.childViewControllers.first as? UINavigationController
+        XCTAssertEqual(screenNavigator?.viewControllers.count, 1,
+                       "there should be only one view controller in the nav stack")
+    }
+    
+    func testNavControllerTypesAfterStartWithoutDocuments() {
+        let rootViewController = coordinator.start(withDocuments: nil)
+        _ = rootViewController.view
+        let screenNavigator = rootViewController.childViewControllers.first as? UINavigationController
+        XCTAssertNotNil(screenNavigator?.viewControllers.first as? CameraViewController,
+                        "first view controller is not a CameraViewController")
+    }
+    
+    func testNavControllerCountAfterStartWithImages() {
+        let capturedImages = [loadImageDocument(withName: "invoice"), loadImageDocument(withName: "invoice2")]
+
+        let rootViewController = coordinator.start(withDocuments: capturedImages)
+        _ = rootViewController.view
+        let screenNavigator = rootViewController.childViewControllers.first as? UINavigationController
+        XCTAssertEqual(screenNavigator?.viewControllers.count, 2,
+                       "there should be 2 view controllers in the nav stack")
+    }
+    
+    func testNavControllerTypesAfterStartWithImages() {
+        let capturedImages = [loadImageDocument(withName: "invoice"), loadImageDocument(withName: "invoice2")]
+
+        let rootViewController = coordinator.start(withDocuments: capturedImages)
+        _ = rootViewController.view
+        let screenNavigator = rootViewController.childViewControllers.first as? UINavigationController
+        
+        XCTAssertNotNil(screenNavigator?.viewControllers.first as? CameraViewController,
```

Comment:
I'm not sure about checking a view controller that's not even visible in screen. It's a conceptual thing, but usually it's better to test only the desired result of what you're doing. Not some side effects. In other words, you should tests your classes kind of as black boxes. Test only the end effect, not the way it is achieved.

We have a lot of work to do to reach that level, but I just wanted to make a note of it here. It seemed to me that some of these tests are too detailed.

Reply:
But then in this specific case, how do you test that both screens are in the navigation stack?